UPCOMING.md

```diff
@@ -33,3 +33,7 @@ dispose() - The runtime is not permitted to dispose the IContainerContext, this
 serviceConfiguration - This property is redundant, and is unused by the runtime. The same information can be found via `deltaManager.serviceConfiguration` on this object if it is necessary.
 
 id - The docId is already logged by the IContainerContext.taggedLogger for telemetry purposes, so this is generally unnecessary for telemetry. If the id is needed for other purposes it should be passed to the consumer explicitly.
+
+## EventForwarder and IDisposable members deprecated from PureDataObject (2023-06-29)
+
+The EventForwarder and IDisposable members have been deprecated from PureDataObject and will be removed in an upcoming release. The EventForwarder pattern was mostly unused by the current implementation, and is also recommended against generallly (instead, register and forward events explicitly). The disposal implementation was incomplete and likely to cause poor behavior as the disposal was not observable by default. Inheritors of the PureDataObject can of course still implement their own disposal logic.
```

Comment:
I haven't looked deeper at implications here - but are we forcing Inheritors to implement IDisposable? If so, what would be the guidance here?

Are we trying to get off of EventForwarder patterns? I see somewhat where small set of use-cases where it may be helpful. If so, can we please create work item.

Reply:
Today inheritors don't get a real IDisposable pattern from PureDataObject (we do nothing for them), so if they do nothing they'll just continue to have the same behavior as today.  If they want to do something particular when the runtime disposes they can, but it's in their hands (both before and after this change).

I'd love for us to get rid of EventForwarder, I see it as an antipattern.  I haven't looked at all the places it's used yet but I'd be generally supportive.  I'll open a work item.

Reply:
Opened AB#4860 and child tasks for this effort.